test/test_bloomerang/test_bloomerang.py

```diff
@@ -0,0 +1,74 @@
+
+    @mock.patch.dict(os.environ, ENV_PARAMETERS)
+    def test_init_env(self):
+        bloomerang = Bloomerang()
+        self.assertEqual(bloomerang.api_key, 'env_api_key')
+        self.assertEqual(bloomerang.client_id, 'env_client_id')
+        self.assertEqual(bloomerang.client_secret, 'env_client_secret')
+
+    @requests_mock.Mocker()
+    def test_authentication(self, m):
+        # API key
+        bloomerang = Bloomerang(api_key='my_key')
+        self.assertEqual(bloomerang.conn.headers['X-API-KEY'], 'my_key')
+
+        # OAuth2
+        m.post(url=bloomerang.uri_auth, json={'code': 'my_auth_code'})
+        m.post(url=bloomerang.uri + 'oauth/token', json={'access_token': 'my_access_token'})
+        bloomerang = Bloomerang(client_id='my_id', client_secret='my_secret')
+        self.assertEqual(bloomerang.authorization_code, 'my_auth_code')
+        self.assertEqual(bloomerang.access_token, 'my_access_token')
+        self.assertEqual(bloomerang.conn.headers['Authorization'], 'Bearer my_access_token')
+
+    def test_base_endpoint(self):
+        url = self.bloomerang._base_endpoint('constituent')
+        self.assertEqual(url, 'https://api.bloomerang.co/v2/constituent/')
+
+        url = self.bloomerang._base_endpoint('constituent', 1234)
+        self.assertEqual(url, 'https://api.bloomerang.co/v2/constituent/1234/')
+
+        url = self.bloomerang._base_endpoint('constituent', '1234')
+        self.assertEqual(url, 'https://api.bloomerang.co/v2/constituent/1234/')
+
+    @requests_mock.Mocker()
+    def test_create_constituent(self, m):
+        m.post(f'{self.bloomerang.uri}constituent/', json=TEST_CONSTITUENT)
+        self.assertEqual(self.bloomerang.create_constituent(), TEST_CONSTITUENT)
+
+    @requests_mock.Mocker()
+    def test_get_constituent(self, m):
+        m.get(f'{self.bloomerang.uri}constituent/{CONSTITUENT_ID}/', json=TEST_CONSTITUENT)
+        self.assertEqual(self.bloomerang.get_constituent(CONSTITUENT_ID), TEST_CONSTITUENT)
+
+    # FAILING TESTS
+
+    @requests_mock.Mocker()
+    def test_update_constituent(self, m):
+        m.put(f'{self.bloomerang.uri}constituent/{CONSTITUENT_ID}/', json=TEST_CONSTITUENT)
+        self.assertEqual(self.bloomerang.update_constituent(CONSTITUENT_ID), TEST_CONSTITUENT)
+
+    @requests_mock.Mocker()
+    def test_delete_constituent(self, m):
+        m.delete(f'{self.bloomerang.uri}constituent/{CONSTITUENT_ID}/', json=TEST_CONSTITUENT_DELETE)
+        self.assertEqual(self.bloomerang.delete_constituent(CONSTITUENT_ID), TEST_CONSTITUENT_DELETE)
+
+    @requests_mock.Mocker()
+    def test_get_constituents(self, m):
+        m.get(f'{self.bloomerang.uri}constituents?skip=0&take=50&id=1|2|3/', json=TEST_CONSTITUENTS)
+        #self.assertEqual(self.bloomerang.get_constituents(CONSTITUENT_ID_LIST), TEST_CONSTITUENTS['Results'])
+        self.assertEqual(Table(self.bloomerang.get_constituents(CONSTITUENT_ID_LIST)), Table(TEST_CONSTITUENTS['Results']))
```

Comment:
@eliotst @shaunagm I posted a question in the Ragtag Parsons channel, but in case it isn't easy to answer it there, I updated my PR to highlight the code giving me issues. 

Line 74 currently fails, and just above it in line 73 you can see the commented out 'passing' version. As you can see, the lines are identical except that the failing version wraps both sides of the assert with `Table`.

Reply:
For connector methods that return multiple objects like `get_constituents`, we want the method to return a `Table`. Does the test pass if you update `get_constituents` to return a `Table` and you wrap `TEST_CONSTITUENTS` in a `Table`.. like:

```
self.assertEqual(self.bloomerang.get_constituents(CONSTITUENT_ID_LIST), Table(TEST_CONSTITUENTS['Results']))
```

Reply:
No, the test does not pass if I do that (that is how I originally tried to implement the method and test, and it results in the same error).

Reply:
Okay. I will take a look this afternoon!

Reply:
Cool, thanks in advance for the help... once this is resolved I should be able to make a final PR for the connector

Reply:
Okay.. I think I got it.. I think that comparison between `Table` objects doesn't work as expected.. so the workaround here is if you a) return a `Table` from `get_constituents`, and b) in the test, pass the result from `get_constituents` (a `Table`) into `list`, you can compare that `list` with `TEST_CONSTITUENTS['Results']`.

So:

```
    @requests_mock.Mocker()
    def test_get_constituents(self, m):
        m.get(f'{self.bloomerang.uri}constituents?skip=0&take=50&id=1|2|3/', json=TEST_CONSTITUENTS)
        #self.assertEqual(self.bloomerang.get_constituents(CONSTITUENT_ID_LIST), TEST_CONSTITUENTS['Results'])
        self.assertEqual(list(self.bloomerang.get_constituents(CONSTITUENT_ID_LIST)), TEST_CONSTITUENTS['Results'])
```

Reply:
Gotcha, I can implement this workaround. I also just discovered another workaround:

The reason I expected I would be able to compare two `Table` objects is that I saw `ActionNetwork().get_people()` comparing two `Table` objects in `test_get_people`. However, the `ActionNetwork` test uses `assert_matching_tables` from `test.utils` rather than `unittest.TestCase.assertEqual` (which is what I'm using).

I found that using `assert_matching_tables` to test for the equality between the `Table` objects works as well.